rfcs/20250430-host-side-scalars/README.md

```diff
@@ -0,0 +1,204 @@
+    large overheads due to re-jitting.
+
+Pros:
+- oneDNN API can already handle this for parameters that do not frequently
+    change.
+
+Cons:
+- Forces user to recreate primitive descriptor and primitive objects every-time
+    that host-side scalar value changes.
+- Potential extra jitting overhead if internal implementation considers that
+    parameter constant.
+
+This option is not recommended if the host-side scalar value is expected to
+frequently change.
+
+### Option 3: Take engine kind in memory descriptor
+
+Here, we would allow users to pass any memory on host or device. This would
+officially tie a given memory descriptor to a device kind.
+
+For compatibility reasons, we could either use `undef` to specify that a memory
+desc will be ties to the same engine kind as the implementation, or introduce a
+new `engine_kind` for that purpose.
+
+A few points need to be made:
+- Mixing host and device memory is already possible through usm shared memory
+    for both sycl and opencl runtime.
+- There is no clear benefit to mix host and device memory, except for the case
+    where the value to pass is a scalar, in which case it can be passed to device
+    kernel as a parameter (instead of passing a pointer).
+- Even if two memories share the same engine kind, it does not guarantee that
+    they will both be accessible from the same context (they might be tied to
+    different engine objects).
+
+Pros:
+- Clear semantic: all memory descriptors are tied to an engine kind.
+
+Cons:
+- Full buffers sharing across devices can already be achieved through runtime
+    mechanisms.
+- No clear benefit for performance other than specific scalar case.
+
+As a result, we don't recommend to specify engine kind for memory descriptors as
+it would likely provide little benefit over host-side scalar specific memory
+descriptor, but being more general, it would also come with more complications.
+
+> Note: Options 3 formally require creating memory objects with a `CPU` engine.
+However, since this is a very specific use case that does not otherwise require
+full CPU support in oneDNN, it is suggested to introduce a new `host` or "null"
+engine. This would allow users to avoid building oneDNN with complete CPU
+support just for this scenario.
+
+## Option 4: Expose new execute function to allow passing scalars.
+
+For both C and C++ APIs, this option would require introducing a new `execute`
+function overload that accepts an additional container (such as `unordered_map`
+or similar) specifically for scalar arguments. This would allow users to pass
+host-side scalars directly as arguments, separate from the usual memory objects.
+
+Pros:
+- Provides a clear and explicit way to pass scalars without overloading the
+    semantics of memory objects.
+- Avoids tying memory descriptors to a specific device or engine.
+- Can be implemented in a backward-compatible manner by adding new overloads.
```

Comment:
I think another important benefit would be getting rid of unnecessary overhead associated with memory descriptor and object creation.